src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -283,6 +283,12 @@ private EventHubAdapterReceiver GetOrCreateReceiver(QueueId queueId)
             return cache;
         }
 
+        protected virtual IEventHubQueueCache CreateCache(IStreamQueueCheckpointer<string> checkpointer, 
```

Comment:
I think in this case, it is better to extract the whole cache factory method into a separate class with its own extension points. For instance, in my case, I want to reuse the logic assigning `AveragingCachePressureMonitor` etc and just extend existing `IEventHubQueueCache` implementation which is `EventHubQueueCache`.